src/applications/vaos/tests/new-appointment/components/CommunityCareProviderSelectionPage/index.unit.spec.jsx

```diff
@@ -302,6 +310,78 @@ describe('VAOS <CommunityCareProviderSelectionPage>', () => {
     );
   });
 
+  it('should display choose provider when remove provider clicked and address is null', async () => {
+    const testState = {
+      facilities: [{ facilityId: '983', isCerner: false }],
+      vapContactInfo: {
+        residentialAddress: null,
+        mailingAddress: {
+          addressLine1: '123 big sky st',
+          city: 'Cincinnati',
+          stateCode: 'OH',
+          zipCode: '45220',
+          latitude: 39.1,
+          longitude: -84.6,
+        },
+      },
+    };
+
+    const updateProfileFieldsExtra = payload => {
+      return {
+        type: 'UPDATE_USER_PROFILE_EXTRA',
+        payload,
+      };
+    };
+
+    const profile = (state = {}, action) => {
+      if (action.type === 'UPDATE_USER_PROFILE_EXTRA') {
+        return Object.assign({}, state, action.payload);
+      }
+
+      return state;
+    };
+
+    const store = createTestStoreWithUserProfile(initialState, profile);
```

Comment:
No need to create the 3 additional constants, `testState`, `updateProfileFieldsExtra` and `profile`. You can achieve the same by using the spread operator on the initialState and traverse down to the key:value pair you want to replace. Simply replace line 344 with below and it will achieve the same thing without having to invoke extra payload or the custom redux store, `createTestStoreWithUserProfile`
```    
     const store = createTestStore({
      ...initialState,
      user: {
        ...initialState.user,
        profile: {
          ...initialState.user.profile,
        },
        vapContactInfo: {
          ...initialState.user.profile.vapContactInfo,
          residentialAddress: null,
        },
      },
    });
   
```

Reply:
@jenniemc - Can we meet in the morning to discuss this. The spread operator on the initialState didn't work when I tried it the other day. The test scaffolding doesn't create the entire state object to be able to clone the object as needed in this test. I think it will be easier to show you in debug.

Reply:
@ptouchton , absolutely. How about after our standup today?